### PR TITLE
feat(waku2): allow using an env variable to set the node's key

### PR DESCRIPTION
**Pull request by richard-ramos**
Required so it's possible to have a fixed peerID to simplify debugging

### CODE REVIEW

File: wakuv2/waku.go [hunk 1]
```diff
@@ -218,13 +219,28 @@ func New(nodeKey string, fleet string, cfg *Config, logger *zap.Logger, appDB *s
...
 	if nodeKey != "" {
 		privateKey, err = crypto.HexToECDSA(nodeKey)
+		if err != nil {
+			return nil, fmt.Errorf("could not convert nodekey into a valid private key: %v", err)
+		}
 	} else {
-		// If no nodekey is provided, create an ephemeral key
-		privateKey, err = crypto.GenerateKey()
-	}
+		nodeKeyStr := os.Getenv("WAKUV2_NODE_KEY")
```
Comment (stefandunca) [inline wakuv2/waku.go]: Why not do this outside, before calling `New`, and pass the key to the `New` function as parameter?

**Review by stefandunca — APPROVED** [review]
Please see my question inline

File: wakuv2/waku.go (hunk 1 above)
Comment (kaichaosun) [reply wakuv2/waku.go]: Should we have a flag to enable such behavior like a dev mode or read env flag, and list it somewhere for easy find out later?

**Review by chaitanyaprem — APPROVED** [review]
LGTM

File: wakuv2/waku.go (hunk 1 above)
Comment (richard-ramos) [reply wakuv2/waku.go]: > @stefandunca : Why not do this outside, before calling New, and pass the key to the New function as parameter?

I put it in the New function for no reason in particular. I'll move this out to outside as suggested.

Comment (stefandunca) [reply wakuv2/waku.go]: >  ... I'll move this out to outside as suggested.

Sounds great, thanks. This should keep the API clean and reduce the responsibility while avoiding side effects due to having three methods to set the same parameter: empty parameter key, key as parameter and env variable.

File: node/status_node_services.go [hunk 2]
```diff
@@ -347,7 +351,29 @@ func (b *StatusNode) wakuV2Service(nodeConfig *params.NodeConfig, telemetryServe
...
-		w, err := wakuv2.New(nodeConfig.NodeKey, nodeConfig.ClusterConfig.Fleet, cfg, logutils.ZapLogger(), b.appDB, b.timeSource(), signal.SendHistoricMessagesRequestFailed, signal.SendPeerStats)
+		var nodeKey *ecdsa.PrivateKey
+		var err error
+		if nodeConfig.NodeKey != "" {
+			nodeKey, err = crypto.HexToECDSA(nodeConfig.NodeKey)
+			if err != nil {
+				return nil, fmt.Errorf("could not convert nodekey into a valid private key: %v", err)
+			}
+		} else {
+			nodeKeyStr := os.Getenv("WAKUV2_NODE_KEY")
```
Comment (Ivansete-status) [inline node/status_node_services.go:362]: Maybe we could use a more concise env var. For example, in `nwaku-compose` we use `NODEKEY`:
https://github.com/waku-org/nwaku-compose/blob/bba6f8b9a5ad7c7758827a5b6109d4916f0c4cc0/.env.example#L15

**Review by Ivansete-status — APPROVED** [review]
LGTM! Just added a nitpick comment :)

**Review by plopezlpz — APPROVED** [review]

**Pull request merged**